cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -120,13 +120,25 @@ func newEtcdClient() (client *etcd.Client) {
 			break
 		}
 	}
-	client = etcd.NewClient([]string{*etcd_server})
-	if client == nil {
-		return nil
+	// loop until we have > 0 machines && machines[0] != ""
+	timeout := false
+	go func() {
+		<-time.After(10 * time.Second)
+		timeout = true
+	}()
+	for !timeout {
+		client = etcd.NewClient([]string{*etcd_server})
+		if client == nil {
+			return nil
+		}
+		client.SyncCluster()
+		machines := client.GetCluster()
+		if len(machines) > 0 && len(machines[0]) > 0 {
+			return client
+		}
 	}
-	client.SyncCluster()
-
-	return client
+	log.Fatal("Timed out waiting for correct response from etcd server")
```

Comment:
I'd suggest making this error message more specific to aid debugging.  Something along the lines of:

"Timed out after xxx seconds waiting for at least 1 synchronized etcd server in the cluster."